process_manager/bin.py

```diff
@@ -0,0 +1,129 @@
+from datetime import datetime, timedelta
+
+from runner import PriorityRunner
+from multiprocessing import Process
+from enum import Enum
+from dataclasses import dataclass
+from heapq import heappop, heappush
+import logging
+import asyncio
+import signal
+
+
+class NoTaskError(Exception):
+    pass
+
+
+class SchedulerTask:
+    def __init__(self,
+                 start_time: datetime,
+                 end_time: datetime,
+                 priority: int,
+                 is_realtime: bool,
+                 target: callable,
+                 timeout: int = 10) -> None:
+
+        self.start_time = start_time
+        self.end_time = end_time
+        self.priority = priority
+        self.is_realtime = is_realtime
+        self.job_id = hash((self.start_time, self.end_time))
+        self.timeout = timeout
+        self.target = target
+        self.process = None
+    
+    def __repr__(self):
+        return f"{self.target} <- {{prio={self.priority}, timeout={self.timeout}}}"
+
+
+class BinType(Enum):
+    REALTIME = 'realtime'
+    STANDARD = 'standard'
+
+
+@dataclass
+class BinConfig:
+    bin_type: BinType
+    start: datetime
+    float_after: timedelta
+    length: timedelta
+    number_threads: int = 2
+    bin_size: int = 5
+
+
+class SchedulingBin:
+    def __init__(self,
+                 start: datetime,
+                 float_after: timedelta,
+                 length: timedelta,
+                 number_threads: int,
+                 bin_size: int) -> None:
+        
+        self.start = start
+        self.float_after = float_after
+        self.length = length
+        self.number_threads = number_threads
+        self.bin_size = bin_size
+        prun = PriorityRunner(self.bin_size)
+        prun.add_done_callback(self._schedule_pending)
+        self.runner = prun
+        self.pending_tasks = []
+        self.accepting = True
+        self.priority_queue = []
+        self.running_tasks = []
+
+    def float_bin(self):
+        self.start += self.float_after
```

Comment:
A little bit of thought might still be needed here on the design side of things. Since science asked that we float daily and week bins constantly, they will require special handling, but this is just a placeholder more to me to clarify that and for us to make sure it is implemented down the road. Right now, this is not a high priority modification.

Maybe we can add a:
```
# TODO: Float daily and weekly bins regularly.
```
just so that it's in the code base.